metrics-operator/controllers/common/providers/dynatrace/client/client_test.go

```diff
@@ -51,7 +51,7 @@ func TestAPIClient(t *testing.T) {
 
 	require.NotNil(t, apiClient)
 
-	resp, err := apiClient.Do(context.TODO(), "/query", http.MethodPost, nil)
+	resp, _, err := apiClient.Do(context.TODO(), "/query", http.MethodPost, nil)
```

Comment:
please also bind the received status to a variable and do an assertion on that (also for the other test cases)